src/HotChocolate/Core/test/Types.Tests/Types/Scalars/TimeSpanTypeTests.cs

```diff
@@ -0,0 +1,327 @@
+
+            // act
+            bool success = timeSpanType
+                .TryDeserialize("PT5", out var deserialized);
+
+            // assert
+            Assert.False(success);
+        }
+
+        [Theory]
+        [InlineData(TimeSpanFormat.Iso8601, "P10675199DT2H48M5.4775807S")]
+        [InlineData(TimeSpanFormat.DotNet, "10675199.02:48:05.4775807")]
+        public void Deserialize_TimeSpan_Max(TimeSpanFormat format, string actualValue)
+        {
+            // arrange
+            var timeSpanType = new TimeSpanType(format);
+            TimeSpan timeSpan = TimeSpan.MaxValue;
+
+            // act
+            var deserializedValue = (TimeSpan)timeSpanType
+                .Deserialize(actualValue);
+
+            // assert
+            Assert.Equal(timeSpan, deserializedValue);
+        }
+
+        [Theory]
+        [InlineData(TimeSpanFormat.Iso8601, "-P10675199DT2H48M5.4775808S")]
+        [InlineData(TimeSpanFormat.DotNet, "-10675199.02:48:05.4775808")]
+        public void Deserialize_TimeSpan_Min(TimeSpanFormat format, string actualValue)
+        {
+            // arrange
+            var timeSpanType = new TimeSpanType(format);
+            TimeSpan timeSpan = TimeSpan.MinValue;
+
+            // act
+            var deserializedValue = (TimeSpan)timeSpanType
+                .Deserialize(actualValue);
+
+            // assert
+            Assert.Equal(timeSpan, deserializedValue);
+        }
+
+        [Fact]
+        public void Deserialize_InvalidString()
+        {
+            // arrange
+            var timeSpanType = new TimeSpanType();
+
+            // act
+            bool success = timeSpanType
+                .TryDeserialize("bad", out object deserialized);
+
+            // assert
+            Assert.False(success);
+        }
+
+        [Fact]
+        public void Deserialize_Null_To_Null()
+        {
+            // arrange
+            var timeSpanType = new TimeSpanType();
+
+            // act
+            bool success = timeSpanType
+                .TryDeserialize(null, out object deserialized);
+
+            // assert
+            Assert.True(success);
+            Assert.Null(deserialized);
+        }
+
+        [Fact]
+        public void ParseLiteral_NullValueNode()
+        {
+            // arrange
+            var timeSpanType = new TimeSpanType();
+            NullValueNode literal = NullValueNode.Default;
+
+            // act
+            object value = timeSpanType.ParseLiteral(literal);
+
+            // assert
+            Assert.Null(value);
+        }
+
+        [Fact]
+        public void ParseValue_Null()
+        {
+            // arrange
+            var timeSpanType = new TimeSpanType();
+
+            // act
+            IValueNode literal = timeSpanType.ParseValue(null);
+
+            // assert
+            Assert.IsType<NullValueNode>(literal);
+        }
+
+        [Fact]
+        public void PureCodeFirst_AutomaticallyBinds_TimeSpan()
```

Comment:
@glucaci I also added some integration tests so that we now verify that the type is automatically bound to System.TimeSpan. I also did an integration test that shows that we can bind two fields with different formats.